3_javascript/javascript1/week2/javascript.js

```diff
@@ -0,0 +1,146 @@
+}
+if(gender === "female"){
+    return `Lady ${firstname} ${surname}`;
+}
+if (gender === ""){
+    return `${firstname} ${surname}`;
+}
+}
+
+const userFormalName = getFullname("Soeren", "Sepehr", "");
+const userFormalName1 = getFullname("Helena", "HME", "female");
+const userFormalName2 = getFullname("Shahram", "S.K.Loo", "male");
+const userFormalName3 = getFullname("Carl-Emil", "R.H.Hansen", "male" );
+
+console.log(userFormalName);
+console.log(userFormalName1);
+console.log(userFormalName2);
+console.log(userFormalName3);
+
+
+console.log("*******     Event application     *******");
+function getEventWeekDay(howManyDaysUntilEvent) {
+    const weekdays = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    const todayDate = new Date();
+    let dayIndex = todayDate.getDay();
+    const eventIndex = (dayIndex + howManyDaysUntilEvent) % 7;
+
+    return weekdays[eventIndex];
+}
+
+console.log(getEventWeekDay(5));
+console.log(getEventWeekDay(11));
+
+//A comment for dear mentor:
+/* The question was very complex.
+Despite my extensive efforts to understand it
+(searching on Google, checking educational websites, watching videos on YouTube,
+looking at others' answers, and seeking help from ChatGPT)
+I still found it challenging to grasp the issue.
+I tried my best to comprehend it but wasn't very successful.
+Please explain the solution to this problem in a simple manner. */
+
+console.log("*******     Weather wear     *******");
+// function outsideTempreture(tempreture) {
+//     const wearInFrazing = "jacket and gloves";
+//     const wearInCold = "sweater";
+//     const wearInWarm = "top and shorts";
+
+//     if(tempreture <= 10) {
+//         return `Frazing...! Wear ${wearInFrazing}.`
+//     }
+//     if(tempreture > 10 && tempreture < 19) {
+//         return `Almost cold...! Wear ${wearInCold}.`
+//     } else {
+//         return `Sun shine...! Wear ${wearInWarm} and remember suncream and sunglasses.`
+//     }
+// }
+// const wearInWarm = outsideTempreture(27);
+// console.log(wearInWarm);
+
+const tempreture = 20;
+switch(tempreture) {
+
+    case(tempreture < 10 || tempreture === 10):
+    console.log("Freezing...! Wear jacket and gloves.");
+    break;
+
+    case(tempreture > 10 && tempreture < 19 || tempreture === 19):
+    console.log("Almost cold...! Wear sweater.");
+    break;
+
+    case(tempreture > 19 && tempreture < 39 || tempreture === 39):
+    console.log("Sun shine...! Wear top, shorts and remember suncream and sunglasses.");
+    break;
+
+    default:
+    console.log("Warning!... Risk of heatstroke");   
+}
+
+//In this case there is some problem. As you can see I have set the temprature on 20:
+//here (const tempreture = 20;). So we should face with this sentence:
+//(Sun shine...! Wear top, shorts and remember suncream and sunglasses.)
+//but we have (Warning!... Risk of heatstroke) in the browser and I cant undrestand what is wrong with that :(
+//please give me a clue!
```

Comment:
The clue is to not use switch case, use for loop instead.
the reason is, whatever type of variable you use in switch, you should use the same in the cases, so:

this should be ok
```suggestion
temprature1 = 20;
switch (temprature1) {
  case 20:
    console.log("this is cold");
}
```
because temprature is a number, and the case is a number
but what you did is:

```suggestion
temprature1 = 20;
switch (temprature1) {
  case temprature===20:
    console.log("this is cold");
}
```
this does not work, because the case temprature===20 is a boolean not a number.

Reply:
oooooh!
Now I understand.

super explanation! I appreciate it